File: Program/Java/MotorSport.java
// Subclass dari Motor (Mengimplementasikan Multilevel Inheritance)
public class MotorSport extends Motor {
    // Atribut spesifik untuk jenis MotorSport
    private int kecepatanMaksimal;  // Kecepatan puncak dalam km/h
    private double harga;            // Harga kendaraan dalam Rupiah
    private String sistemPengereman; // Contoh: ABS, Dual Disc, Brembo, dll.

    // Constructor memanggil constructor milik kelas Motor
    public MotorSport(int id, String merk, int tahun, int kapasitasMesin, String jenisTransmisi, double kapasitasTangki, int kecepatanMaksimal, double harga, String sistemPengereman) {
        super(id, merk, tahun, kapasitasMesin, jenisTransmisi, kapasitasTangki);
        this.kecepatanMaksimal = kecepatanMaksimal;
        this.harga = harga;
        this.sistemPengereman = sistemPengereman;
    }

    // Getter dan Setter untuk Kecepatan Maksimal
    public void setKecepatanMaksimal(int kecepatanMaksimal) { this.kecepatanMaksimal = kecepatanMaksimal; }
    public int getKecepatanMaksimal() { return kecepatanMaksimal; }

    // Getter dan Setter untuk Harga
    public void setHarga(double harga) { this.harga = harga; }
    public double getHarga() { return harga; }

    // Getter dan Setter untuk Sistem Pengereman
    public void setSistemPengereman(String sistemPengereman) { this.sistemPengereman = sistemPengereman; }
    public String getSistemPengereman() { return sistemPengereman; }
}

File: Program/python/MotorSport.py
from Motor import Motor

# Subclass dari Motor (Multilevel Inheritance)
class MotorSport(Motor):
    def __init__(self, id_kendaraan=0, merk="", tahun=0, kapasitas_mesin=0, jenis_transmisi="", kapasitas_tangki=0.0, kecepatan_maksimal=0, harga=0.0, sistem_pengereman=""):
        # Memanggil constructor milik class Motor
        super().__init__(id_kendaraan, merk, tahun, kapasitas_mesin, jenis_transmisi, kapasitas_tangki)
        self._kecepatan_maksimal = kecepatan_maksimal
        self._harga = harga
        self._sistem_pengereman = sistem_pengereman

    # Getter dan Setter untuk Kecepatan Maksimal
    def get_kecepatan_maksimal(self):
        return self._kecepatan_maksimal

    def set_kecepatan_maksimal(self, kecepatan_maksimal):
        self._kecepatan_maksimal = kecepatan_maksimal

    # Getter dan Setter untuk Harga
    def get_harga(self):
        return self._harga

    def set_harga(self, harga):
        self._harga = harga

    # Getter dan Setter untuk Sistem Pengereman
    def get_sistem_pengereman(self):
        return self._sistem_pengereman

    def set_sistem_pengereman(self, sistem_pengereman):
        self._sistem_pengereman = sistem_pengereman